windows/shared/logging/logging-plog-windows/CMakeLists.txt

```diff
@@ -19,6 +19,7 @@ target_include_directories(logging-plog-windows
 target_link_libraries(logging-plog-windows
     PUBLIC
         plog::plog
+        logutils

```

Comment:
Why does this need to be `PUBLIC` instead of `PRIVATE`? Does some public part of this target/lib use something from `logutils`? The only public part of `logutils` seems to be `GetLogName()`, which I didn't notice being used here in the public interface of `logging-plog-windows`.

Reply:
Right now I have logging-plog-windows providing the tracelogging stuff as well as ```GetLogName```. I originally had it also provide the debug wrapper but I removed it, so now I guess I could have it not provide ```GetLogName`` and have all the windows code explicitly link with both the logutils library and the logging-plog-windows library. However, we might want to add something else to the windows specific logging library in the future so I think we should leave this structure.

Reply:
> Right now I have logging-plog-windows providing the tracelogging stuff as well as GetLogName.

The tracelogging stuff, AFAIK, doesn't have anything in its public interface that uses anything from `logutils`. In that case, it should be linked as `PRIVATE`. If this changes later, then we can change to `PUBLIC` is appropriate.

> have all the windows code explicitly link with both the logutils library and the logging-plog-windows library

We should never have to explicitly link to any indirect dependencies; as long as we specify the link visibility (as here) correctly, CMake will take care of the rest. Otherwise, it becomes untenable figuring out all the dependencies in the chain of each lib.

Reply:
The tracelogging stuff also doesn't use the logutils library at all. I put it in PUBLIC so that linking against ```logging-plog-windows``` would provide both the logutils stuff and the tracelogging stuff.

Reply:
I understand what the intention is, but `target_link_libraries` has a specific meaning: it transitively propagates the dependencies of the target. In this case, you're including another target as `PUBLIC` which declares a certain relationship, yet that relationship isn't reflective of the program structure since the target doesn't actually depend on `logutils`. Looking at that in isolation, it's confusing.

Despite that, if you're going to keep this, then I suggest adding a comment to clarify why it's being added as a `PUBLIC` transitive dependent when it's not actually a dependency so there is no confusion.

Reply:
I see, is there a way around this so it's not confusing, or is adding a comment the standard way?

Reply:
I don't know of any way it's not confusing, but I've also not seen this done before (adding a non-dependent in the target link libraries). The standard approach is to only add direct, known dependencies to the target link libraries of a target.

Reply:
got it. In that case, I'll make the tracelogging stuff it's own library